crates/state/src/batch.rs

```diff
@@ -102,178 +128,156 @@ impl From<SignedBatchCheckpoint> for BatchCheckpoint {
     }
 }
 
+/// Contains metadata describing a batch checkpoint, including the L1 and L2 height ranges
+/// it covers and the final L2 block ID in that range.
 #[derive(
     Clone, Debug, Eq, PartialEq, Arbitrary, BorshDeserialize, BorshSerialize, Deserialize, Serialize,
 )]
 pub struct BatchInfo {
-    /// The index of the checkpoint
-    pub idx: u64,
+    /// Checkpoint epoch
+    pub epoch: u64,
 
     /// L1 height range(inclusive) the checkpoint covers
-    pub l1_range: (u64, u64),
+    pub l1_range: (L1BlockCommitment, L1BlockCommitment),
 
     /// L2 height range(inclusive) the checkpoint covers
-    pub l2_range: (u64, u64),
+    pub l2_range: (L2BlockCommitment, L2BlockCommitment),
 
+    /// The last L2 block upto which this checkpoint covers since the previous checkpoint
+    pub l2_blockid: L2BlockId,
```

Comment:
This is no longer necessary to have here as a separate field since it's in `l2_range.1`.